### PR TITLE
Simplify message handlers logic

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: lib/freddy/message_handler_adapaters.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      def for(delivery)
+        if delivery.type == 'request'
+          @request_handler
+        else
+          @standard_message_handler
+        end
+      end
+    end
+
+    class StandardMessageHandler
```
Comment (willmore) [inline lib/freddy/message_handler_adapaters.rb]: What does this class do?

Comment (indrekj) [inline lib/freddy/message_handler_adapaters.rb]: Lets assume you have a responder:
```ruby
freddy.respond_to 'SomeQueue' do |payload, msg_handler|
  msg_handler.success(this: 'is a response')
end
```

There's two ways how to send messages to it:
`response = freddy.deliver_with_response(..)`
`freddy.deliver(...)`

The first one expects a response, the seconds one does not. So for the first message we'll use `RequestHandler` and for the second one we this handler (`StandardMessageHandler`) which does a no-op operation on `#success` call.

Comment (indrekj) [inline lib/freddy/message_handler_adapaters.rb]: Does this make sense? I can rename this class to something more meaningful. Mabye `NoopHandler` or something?

Comment (willmore) [inline lib/freddy/message_handler_adapaters.rb]: Makes sens - `NoOpHandler` would communicate the class' behavior better.

**Comment by indrekj** [comment]
Updated

**Comment by stenver** [comment]
+1

**Pull request merged**